src/services/component-descriptor-service.ts

```diff
@@ -0,0 +1,45 @@
+import { BundleDescriptor } from '../models/bundle-descriptor'
+import { Component, ComponentType } from '../models/component'
+import BundleDescriptorService from './bundle-descriptor-service'
+
+export class ComponentDescriptorService {
```

Comment:
As the name indicates, this class is responsible only for a single component (e.g. getting its version). You put here other operations that should not be responsibility of this class. I would create instead another service class named `ComponentService`, that will be using `BundleDescriptorService` and `ComponentDescriptorService`